build: release v6.24.0

timotheeg opened the pull request:
## New

- build: merge release 6.23.2 back to develop [`#5606`](https://github.com/opengovsg/FormSG/pull/5606)
- build: Release v6.23.2 hotfix [`#5602`](https://github.com/opengovsg/FormSG/pull/5602)
- test: fix race condition in field schema test [`#5604`](https://github.com/opengovsg/FormSG/pull/5604)
- fix: avoid using merge to set query after reordering fields [`#5605`](https://github.com/opengovsg/FormSG/pull/5605)
- feat: push static assets to s3 [`#5595`](https://github.com/opengovsg/FormSG/pull/5595)
- fix: return 404 for unmatched static assets [`#5579`](https://github.com/opengovsg/FormSG/pull/5579)
- chore: merge v6.23.1 to develop [`#5594`](https://github.com/opengovsg/FormSG/pull/5594)
- test: fix selectors for tests [`#5597`](https://github.com/opengovsg/FormSG/pull/5597)
- fix: fix hot reload not working due to missing `process` variable on React [`#5547`](https://github.com/opengovsg/FormSG/pull/5547)
- fix: remove unecessary import [`#5576`](https://github.com/opengovsg/FormSG/pull/5576)
- feat: upgrade axios to 1.2.1 [`#5568`](https://github.com/opengovsg/FormSG/pull/5568)
- chore: merge v6.23.0 into develop  [`#5574`](https://github.com/opengovsg/FormSG/pull/5574)
- chore: bump version to 6.23.2 [`1256707`](https://github.com/opengovsg/FormSG/commit/1256707db95a30486c2ccfc731d9ff6cbdb7dc40)

## Dependencies

- fix(deps): bump sqs-producer from 2.1.0 to 3.1.0 [`#5590`](https://github.com/opengovsg/FormSG/pull/5590)
- fix(deps): bump @aws-sdk/client-cloudwatch-logs from 3.218.0 to 3.234.0 [`#5599`](https://github.com/opengovsg/FormSG/pull/5599)
- fix(deps): bump libphonenumber-js from 1.10.14 to 1.10.15 [`#5553`](https://github.com/opengovsg/FormSG/pull/5553)
- fix(deps): bump moment-timezone from 0.5.39 to 0.5.40 [`#5586`](https://github.com/opengovsg/FormSG/pull/5586)
- fix(deps): bump zod from 3.19.1 to 3.20.2 in /shared [`#5562`](https://github.com/opengovsg/FormSG/pull/5562)
- fix(deps): bump type-fest from 3.3.0 to 3.4.0 in /shared [`#5563`](https://github.com/opengovsg/FormSG/pull/5563)
- fix(deps): bump formsg-javascript-sdk from 0.9.0 to 0.10.0 [`#5578`](https://github.com/opengovsg/FormSG/pull/5578)

## Dev-Dependencies

- chore(deps-dev): bump regenerator from 0.14.9 to 0.14.10 [`#5608`](https://github.com/opengovsg/FormSG/pull/5608)
- chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.46.1 to 5.47.0 in /shared [`#5592`](https://github.com/opengovsg/FormSG/pull/5592)
- chore(deps-dev): bump lint-staged from 13.0.3 to 13.1.0 [`#5525`](https://github.com/opengovsg/FormSG/pull/5525)
- chore(deps-dev): bump @typescript-eslint/parser from 5.46.1 to 5.47.0 in /shared [`#5593`](https://github.com/opengovsg/FormSG/pull/5593)
- chore(deps-dev): bump @playwright/test from 1.27.1 to 1.29.0 [`#5585`](https://github.com/opengovsg/FormSG/pull/5585)

timotheeg committed:
Merge pull request #5574 from opengovsg/release-al2

chore: merge v6.23.0 into develop

mantariksh committed:
feat: upgrade axios to 1.2.1 (#5568)

* feat: upgrade axios to latest version

* fix: try upgrading jest to latest

* fix: add required peer dep

* fix: upgrade ts-jest

* fix: upgrade jest types to match jest version

* test: replace all mocked with jest.mocked

* fix: fix mocked import in frontend tests

* test: add required arg in rate limit tests

timotheeg committed:
fix: remove unecessary import (#5576)

lint: remove unecessary import

karrui committed:
fix(deps): bump formsg-javascript-sdk from 0.9.0 to 0.10.0 (#5578)

* feat: use specific commit version of formsg-sdk

* fix(deps): bump formsg-javascript-sdk from 0.9.0 to 0.10.0

dependabot[bot] committed:
fix(deps): bump type-fest from 3.3.0 to 3.4.0 in /shared (#5563)

Bumps [type-fest](https://github.com/sindresorhus/type-fest) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/sindresorhus/type-fest/releases)
- [Commits](https://github.com/sindresorhus/type-fest/compare/v3.3.0...v3.4.0)

---
updated-dependencies:
- dependency-name: type-fest
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
fix(deps): bump zod from 3.19.1 to 3.20.2 in /shared (#5562)

Bumps [zod](https://github.com/colinhacks/zod) from 3.19.1 to 3.20.2.
- [Release notes](https://github.com/colinhacks/zod/releases)
- [Changelog](https://github.com/colinhacks/zod/blob/master/CHANGELOG.md)
- [Commits](https://github.com/colinhacks/zod/compare/v3.19.1...v3.20.2)

---
updated-dependencies:
- dependency-name: zod
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @playwright/test from 1.27.1 to 1.29.0 (#5585)

Bumps [@playwright/test](https://github.com/Microsoft/playwright) from 1.27.1 to 1.29.0.
- [Release notes](https://github.com/Microsoft/playwright/releases)
- [Commits](https://github.com/Microsoft/playwright/compare/v1.27.1...v1.29.0)

---
updated-dependencies:
- dependency-name: "@playwright/test"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
fix(deps): bump moment-timezone from 0.5.39 to 0.5.40 (#5586)

Bumps [moment-timezone](https://github.com/moment/moment-timezone) from 0.5.39 to 0.5.40.
- [Release notes](https://github.com/moment/moment-timezone/releases)
- [Changelog](https://github.com/moment/moment-timezone/blob/develop/changelog.md)
- [Commits](https://github.com/moment/moment-timezone/compare/0.5.39...0.5.40)

---
updated-dependencies:
- dependency-name: moment-timezone
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
fix(deps): bump libphonenumber-js from 1.10.14 to 1.10.15 (#5553)

Bumps [libphonenumber-js](https://gitlab.com/catamphetamine/libphonenumber-js) from 1.10.14 to 1.10.15.
- [Release notes](https://gitlab.com/catamphetamine/libphonenumber-js/tags)
- [Changelog](https://gitlab.com/catamphetamine/libphonenumber-js/blob/master/CHANGELOG.md)
- [Commits](https://gitlab.com/catamphetamine/libphonenumber-js/compare/v1.10.14...v1.10.15)

---
updated-dependencies:
- dependency-name: libphonenumber-js
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/parser from 5.46.1 to 5.47.0 in /shared (#5593)

chore(deps-dev): bump @typescript-eslint/parser in /shared

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.46.1 to 5.47.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.47.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

karrui committed:
fix: fix hot reload not working due to missing `process` variable on React (#5547)

timotheeg committed:
Merge remote-tracking branch 'origin/release-al2' into develop

dependabot[bot] committed:
chore(deps-dev): bump lint-staged from 13.0.3 to 13.1.0 (#5525)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 13.0.3 to 13.1.0.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v13.0.3...v13.1.0)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

justynoh committed:
test: fix selectors for tests (#5597)

* test: fix selectors for tests

* test: fix more selectors, fix tests

timotheeg committed:
Merge remote-tracking branch 'origin/develop' into backport_v6.23.1_to_develop

timotheeg committed:
Merge pull request #5594 from opengovsg/backport_v6.23.1_to_develop

chore: merge v6.23.1 to develop

timotheeg committed:
fix: return 404 for unmatched static assets (#5579)

* fix: return 404 for unmatched static assets

* fix: add comment for root files 404

* ref: improve comment for react static route

Co-authored-by: tshuli <63710093+tshuli@users.noreply.github.com>

* ref: bring consistency to route error handlers

* fix: looser regex for root files 404

* fix: fix invalid imports for error handler middleware

Co-authored-by: tshuli <63710093+tshuli@users.noreply.github.com>

dependabot[bot] committed:
fix(deps): bump @aws-sdk/client-cloudwatch-logs from 3.218.0 to 3.234.0 (#5599)

Bumps [@aws-sdk/client-cloudwatch-logs](https://github.com/aws/aws-sdk-js-v3/tree/HEAD/clients/client-cloudwatch-logs) from 3.218.0 to 3.234.0.
- [Release notes](https://github.com/aws/aws-sdk-js-v3/releases)
- [Changelog](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-cloudwatch-logs/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-js-v3/commits/v3.234.0/clients/client-cloudwatch-logs)

---
updated-dependencies:
- dependency-name: "@aws-sdk/client-cloudwatch-logs"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tshuli committed:
feat: push static assets to s3 (#5595)

* feat: upload static assets to s3

* feat: add static asset URL for craco

* chore: did not work, try injecting PUBLIC_URL to .env instead

* feat: copy frontend folder contents as well

* feat: include static asset bucket in CSP headers

* chore: remove unnecessary STATIC_ASSET_URL env var

* feat: add manifestSrc to CSP

* fix: typing

* chore: add trailing backslash to csp

* chore: add config.aws.staticAssetsBucketUrl to workerSrc CSP

* chore: undo changes to csp, do not set craco PUBLIC_URL

* feat: reverse proxy middleware attempt to fetch from s3 bucket

* feat: load env var for STATIC_ASSETS_S3_BUCKET

* fix: extra slash, destructure status

* chore: axios error handling and logging

* docs: log error from s3

dependabot[bot] committed:
fix(deps): bump sqs-producer from 2.1.0 to 3.1.0 (#5590)

Bumps [sqs-producer](https://github.com/bbc/sqs-producer) from 2.1.0 to 3.1.0.
- [Release notes](https://github.com/bbc/sqs-producer/releases)
- [Commits](https://github.com/bbc/sqs-producer/commits/v3.1.0)

---
updated-dependencies:
- dependency-name: sqs-producer
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.46.1 to 5.47.0 in /shared (#5592)

chore(deps-dev): bump @typescript-eslint/eslint-plugin in /shared

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.46.1 to 5.47.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.47.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

justynoh committed:
fix: avoid using merge to set query after reordering fields (#5605)

fix: avoid using merge

mantariksh committed:
test: fix race condition in field schema test (#5604)

* test: fix race condition in field schema test

* refactor: await and test each field sequentially

* refactor: use object key shorthand

dependabot[bot] committed:
chore(deps-dev): bump regenerator from 0.14.9 to 0.14.10 (#5608)

Bumps [regenerator](https://github.com/facebook/regenerator) from 0.14.9 to 0.14.10.
- [Release notes](https://github.com/facebook/regenerator/releases)
- [Commits](https://github.com/facebook/regenerator/compare/regenerator@0.14.9...regenerator@0.14.10)

---
updated-dependencies:
- dependency-name: regenerator
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

timotheeg committed:
Merge pull request #5606 from opengovsg/release-al2

build: merge release 6.23.2 back to develop

timotheeg commented:
## Tests - feat: upgrade axios to 1.2.1 [`#5568`](https://github.com/opengovsg/FormSG/pull/5568)

@justynoh 

We use `axios` for:
- Every single API call from frontend to backend
- Verifying reCAPTCHA responses through Google reCAPTCHA API
- Sending webhooks

I wrote these tests to try to test all the ways in which we might be using `axios`.

**GET**
- [ ] Load admin dashboard

**PUT**
- [ ] Create form logic. Make an update to the form logic you just created and check that it goes through correctly.

**POST (application/json)**
- [ ] Create a form field

**POST (multipart/form-data)**
- [ ] Make a form submission and check that it goes through 

**POST (with query params)**
- [ ] Reorder form fields

**PATCH**
- [ ] Make an update to form settings and check that settings are updated correctly.

**DELETE**
- [ ] Archive a form

**Interceptors**
This test relies on buggy behaviour because I couldn't think of any other easy way to test this.
- [ ] Log in to admin dashboard. In a separate tab, log in to a MyInfo form. Remove the `MyInfoCookie` cookie using developer tools. Submit the form and get a 401. Switch back to admin dashboard tab and check that you are logged out.

**reCAPTCHA**
- [ ] Check that you can submit a form with recaptcha, assuming you are not a robot.

**Webhooks**
- [ ] Send a webhook successfully, check that it is logged correctly in FormSG logs ("Webhook POST succeeded")
- [ ] Send a webhook which fails ("Webhook POST failed")

mantariksh commented:
## `sqs-producer` major version bump
- [ ] Enable webhook retries on a form. Send a webhook to a server which returns a non-200 response. Check the logs for a `Pushed webhook to queue` message. Wait for between 4-6min for the next attempt to come in, and return a non-200 response again. Check the logs for another `Pushed webhook to queue` message.

dependabot[bot] committed:
fix(deps): bump @sentry/browser from 7.21.1 to 7.28.0 (#5607)

Bumps [@sentry/browser](https://github.com/getsentry/sentry-javascript) from 7.21.1 to 7.28.0.
- [Release notes](https://github.com/getsentry/sentry-javascript/releases)
- [Changelog](https://github.com/getsentry/sentry-javascript/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-javascript/compare/7.21.1...7.28.0)

---
updated-dependencies:
- dependency-name: "@sentry/browser"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @types/nodemailer from 6.4.6 to 6.4.7 (#5609)

Bumps [@types/nodemailer](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/nodemailer) from 6.4.6 to 6.4.7.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/nodemailer)

---
updated-dependencies:
- dependency-name: "@types/nodemailer"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
fix(deps): bump sqs-consumer from 5.7.0 to 6.1.0 (#5591)

* fix(deps): bump sqs-consumer from 5.7.0 to 6.1.0

Bumps [sqs-consumer](https://github.com/BBC/sqs-consumer) from 5.7.0 to 6.1.0.
- [Release notes](https://github.com/BBC/sqs-consumer/releases)
- [Commits](https://github.com/BBC/sqs-consumer/compare/v5.7.0...v6.1.0)

---
updated-dependencies:
- dependency-name: sqs-consumer
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* fix: remove custom SQS instance

sqs-consumer now uses v3 of the AWS SDK, which reuses TCP connections
by default, so there is no need to provide a custom SQS instance.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Antariksh <antarikshmahajan@gmail.com>

wanlingt committed:
ci: separate e2e and backend ci flows (#5483)

* ci: separate e2e and backend ci flows

* ci: shard out build step and use build artifact

* ci: comment out conditional for backend and e2e for testing

* ci: make directory for artifact

* fix: remove retention-days from upload-artifact

* fix: add build files

* fix: download build file to current working directory

* fix: add package*.json to build output

* fix: run backend and e2e tests conditionally

* feat: remove retention days for build artifact

* fix: run backend and e2e tests conditionally

tshuli committed:
Merge pull request #5614 from opengovsg/release-al2

build: merge 6.23.3 into develop

Commit:
chore: bump version to v6.24.0
